### DISCUSSION TITLE
Inspiration

### DISCUSSION THREAD

**Discussion by kleenkanteen** [komoot/photon#801] [General]
Why did you make this project when OSM's Nominatim exists?: https://github.com/osm-search/Nominatim

**Comment by kleenkanteen** [comment]
@lonvia forgot to ping

**Comment by kleenkanteen** [comment]
I came across this article, the search-as-you-type feature is neat: https://www.geoapify.com/nominatim-vs-photon-geocoder

How photon handles something vs. Nominatim: https://github.com/komoot/photon/issues/357

**Comment by mtmail** [comment]
The company https://www.komoot.de/ needed a geocoder on village/city/state/region level which supported type-ahead/autocomplete. They built their own because most existing solutions including Nominatim didn't offer it. It was open source and ran free on photon.komoot.de. @lonvia is the current maintainer, she's also the maintainer of Nominatim. Photon imports data from Nominatim so the projects benefit from each other. As you saw from the geoafiy article the APIs are slightly different, installation methods differ, so do data sizes, country and language coverage, input data, customization options. I don't think one was inspired by the other necessarily. There are dozens of search engines and software dedicated to geocoding and their APIs are often similar.
